upup/pkg/fi/fitasks/keypair.go

```diff
@@ -44,6 +45,9 @@ type Keypair struct {
 	Type string `json:"type"`
 	// LegacyFormat is whether the keypair is stored in a legacy format.
 	LegacyFormat bool `json:"oldFormat"`
+
+	certificate                *fi.TaskDependentResource
```

Comment:
What is this for? It doesn't appear to be read by anything.

Reply:
It just felt more intuitive to store it, as we're also storing the fingerprint.

Reply:
I tend to be more YAGNI on these sorts of things. Anything that's going to need the cert will probably just pull it (and possibly the private key) out of the keystore. But I suppose it doesn't block.